smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ClientConfigGenerator.kt

```diff
@@ -149,51 +144,19 @@ class ClientConfigGenerator(
         }
     }
 
-    private fun renderJavaBuilderInterface() {
+    private fun renderBuilder() {
         ctx.writer.write("")
-            .withBlock("interface FluentBuilder {", "}") {
-                props
-                    .filter { it.propertyType !is ClientConfigPropertyType.ConstantValue }
-                    .forEach { prop ->
-                        // we want the type names sans nullability (?) for arguments
-                        write("fun #1L(#1L: #2L): FluentBuilder", prop.propertyName, prop.symbol.name)
-                    }
-                write("fun build(): #configClass.name:L")
-            }
-    }
-
-    private fun renderDslBuilderInterface() {
-        ctx.writer.write("")
-            .withBlock("interface DslBuilder {", "}") {
-                props
-                    .filter { it.propertyType !is ClientConfigPropertyType.ConstantValue }
-                    .forEach { prop ->
-                        prop.documentation?.let { ctx.writer.dokka(it) }
-                        write("var #L: #P", prop.propertyName, prop.symbol)
-                        write("")
-                    }
-            }
-    }
-
-    private fun renderBuilderImpl() {
-        ctx.writer.write("")
-            .withBlock("internal class BuilderImpl() : FluentBuilder, DslBuilder {", "}") {
+            .withBlock("class Builder {", "}") {
                 // override DSL properties
                 props
                     .filter { it.propertyType !is ClientConfigPropertyType.ConstantValue }
                     .forEach { prop ->
-                        write("override var #L: #D", prop.propertyName, prop.symbol)
+                        write("var #L: #D", prop.propertyName, prop.symbol)
                     }
                 write("")
 
                 write("")
-                write("override fun build(): #configClass.name:L = #configClass.name:L(this)")
-                props
-                    .filter { it.propertyType !is ClientConfigPropertyType.ConstantValue }
-                    .forEach { prop ->
-                        // we want the type names sans nullability (?) for arguments
-                        write("override fun #1L(#1L: #2L): FluentBuilder = apply { this.#1L = #1L }", prop.propertyName, prop.symbol.name)
-                    }
+                write("fun build(): #configClass.name:L = #configClass.name:L(this)")
```

Comment:
question

should `build()` be generated as `internal`? At least until we are sure we need it to be public. If it's public it's accessible from within the DSL builder:

```kotlin
Foo {
    bar = "baz"
    ...
    build()
}
```

Reply:
it should definitely be `internal` since the default constructor is marked `internal`